core/engine/engine.go

```diff
@@ -197,10 +201,79 @@ func (e *Engine) GetLatestAdv(ctx context.Context) (cid.Cid, schema.Advertisemen
 	return latestAdv, ad, nil
 }
 
-func (e *Engine) publishAdvForIndex(ctx context.Context, dealID cid.Cid, metadata []byte, isRm bool) (cid.Cid, error) {
-	// TODO: We should probably prevent providers from being able to advertise
-	// the same index several times. It may lead to a lot of duplicate retrievals?
-	latestAdvID, err := e.getLatest(false)
+// Linksystem used to generate links from a list of cids without
+// persisting anything in the process.
+func noStoreLinkSystem() ipld.LinkSystem {
+	lsys := cidlink.DefaultLinkSystem()
+	lsys.StorageWriteOpener = func(lctx ipld.LinkContext) (io.Writer, ipld.BlockWriteCommitter, error) {
+		buf := bytes.NewBuffer(nil)
+		return buf, func(lnk ipld.Link) error {
+			return nil
+		}, nil
+	}
+	return lsys
+}
```

Comment:
@warpfork is this something that should be a default stub in ipld-prime?

Reply:
I actually thought that `cidlink.DefaultSystem()` did the trick if now `StorageWriterOpener` was set. Maybe we can follow this approach.